src/main/scala/is/hail/expr/ir/PruneDeadFields.scala

```diff
@@ -1006,8 +1006,9 @@ object PruneDeadFields {
         case ta: TArray =>
           val ra = rType.asInstanceOf[TArray]
           val uid = genUID()
-          val ref = Ref(uid, ta.elementType)
+          val ref = Ref(uid, -ta.elementType)
```

Comment:
could `ra` ever have a required element type?

Reply:
Not given our current system. It would totally be possible for somebody to introduce requiredness in a way that breaks this, but we only use requiredness for PLs and dosages right now.